### DISCUSSION TITLE
How to run a user script after loading the page

### DISCUSSION THREAD

**Discussion by marco-eckstein** [gildas-lormeau/single-file-cli#119] [Show and tell]
There is the option `browser-script`, described as "Path of a script executed in the page (and all the frames) before it is loaded". Indeed, my script (JavaScript) is apparently not run _after_ the page has loaded. I don't know what "executed _before_ it is loaded" even means. However, using a little trick still allows me to execute code after the page has loaded. E.g. for YouTube, I do this:

```javascript
const interval = setInterval(() => {
    const element = document.getElementById("expand");
    element.click();
    clearInterval(interval);
}, 50);
```

The callback only reaches its last line if both `document` and `element` are available.

Note that there is the option `user-script-enabled`, which is _true_ by default and described as "Enable the event API allowing to execute scripts before the page is saved". So maybe there is a more straightforward way to run a user script after loading the page, but I couldn't find it.

**Comment by gildas-lormeau** [comment]
The script is run before the page is parsed in fact. To fix the issue, you can listen to the `load` event on the `window` object before running the script, or integrate more deeply with SingleFile as documented here: https://github.com/gildas-lormeau/SingleFile/wiki/How-to-execute-a-user-script-before-a-page-is-saved

**Reply by marco-eckstein** [reply]
Thanks! This works:

```javascript
window.addEventListener("load", () => {
    document.getElementById("expand").click();
});
```
